Source/ASDisplayNode+Layout.mm

```diff
@@ -82,21 +82,19 @@ - (ASLayout *)layoutThatFits:(ASSizeRange)constrainedSize parentSize:(CGSize)par
   }
 
   ASLayout *layout = nil;
-  if (_calculatedDisplayNodeLayout->isValidForConstrainedSizeParentSize(constrainedSize, parentSize)) {
+  NSUInteger version = _layoutVersion;
+  if (_calculatedDisplayNodeLayout != nullptr && _calculatedDisplayNodeLayout->isValid(constrainedSize, parentSize, version)) {
```

Comment:
I made this change so that the redundancy between the first condition and the second condition is more explicit, and the fact that `_calculated` is never null isn't widely known (at least by this guy) and probably not ideal! Why should we have two ways of representing a null display node layout – literally `nullptr` vs this blank entry we create `ASDisplayNodeLayout()`? I'm sensitive to the length argument but I'm going to keep this change because I think the consistency of treatment outweighs the length.

That said I completely agree about the latter! Your suggestion is good or bundling those extra ivars up into a C-struct and storing two props in node `_pendingLayout: ASLayout, _pendingLayoutExtras: ASLayoutMeta` and mutating them together or something. Currently the use of shared_ptrs here is hard on the eyes.